### DISCUSSION TITLE
[Bug] Enum flags - with syntax in record types not allowing to add

### DISCUSSION THREAD

**Discussion by stoepa** [dotnet/csharplang#6665] [Language Ideas] [answered]
Hi all,

Not sure if I might be missing something, but see the following scenario does not work, and not sure if it was missed in the Lang?

```
public record Party(Guid Id, string Name, PartyType PartyType);

[Flags]
public enum PartyType
{
      Good = 0,
      VeryGood = 1,
      Exciting = 2
}

var party = new Party(Guid.NewGuid(), "Halloween", PartyType.Good);

party = party with
{
     PartyType |= PartyType.Exciting
}
```

I know with normal classes this works.

The compiler error is [CS0747](https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CS0747))

**Comment by TahirAhmadov** [comment, marked as answer]
https://github.com/dotnet/csharplang/issues/5176
